src/ansys/geometry/core/shapes/curves/trimmed_curve.py

```diff
@@ -154,6 +156,90 @@ def intersect_curve(self, other: "TrimmedCurve") -> list[Point3D]:
             for point in res.points
         ]
 
+    def transformed_copy(self, matrix: Matrix44) -> "TrimmedCurve":
+        """Return a copy of the trimmed curve transformed by the given matrix.
+
+        Parameters
+        ----------
+        matrix : Matrix44
+            Transformation matrix to apply to the curve.
+
+        Returns
+        -------
+        TrimmedCurve
+            A new trimmed curve with the transformation applied.
+        """
+        transformed_geometry = self.geometry.transformed_copy(matrix)
+        transformed_start = self.start.transform(matrix)
+        transformed_end = self.end.transform(matrix)
+
+        return TrimmedCurve(
+            transformed_geometry,
+            transformed_start,
+            transformed_end,
+            self.interval,
+            self.length,
+        )
+
+    def translate(self, direction: Vector3D, distance: Real | Quantity | Distance) -> None:
```

Comment:
direction should be a UnitVector3D. A Vector3D could be allowed but in that case we should inform the user that a normalization to that direction vector will take place